src/Images/Pexels.php

```diff
@@ -47,17 +108,72 @@ public function get_images( $ai_connection, $token, $business_description ) {
 
-		$response = $ai_connection->fetch_ai_response( $token, $prompt );
+		$response = $ai_connection->fetch_ai_response( $token, $prompt, 30 );
 
-		if ( is_wp_error( $response ) || ! isset( $response['completion'] ) ) {
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		if ( ! isset( $response['completion'] ) ) {
 			return new \WP_Error( 'search_term_definition_failed', __( 'The search term definition failed.', 'woo-gutenberg-products-block' ) );
 		}
 
 		return $response['completion'];
 	}
 
+	/**
+	 * Refine the results returned by Pexels API.
+	 *
+	 * @param  Connection $ai_connection  The AI connection.
+	 * @param  string     $token  The JWT token.
+	 * @param  string     $business_description  The business description.
+	 * @param  array      $returned_images  The returned images.
+	 *
+	 * @return array|\WP_Error The refined images, or WP_Error if the request failed.
+	 */
+	private function refine_returned_images_results( $ai_connection, $token, $business_description, $returned_images ) {
+		$image_titles = array();
+		foreach ( $returned_images as $returned_image ) {
+			if ( isset( $returned_image['title'] ) ) {
+				$image_titles[] = $returned_image['title'];
+			}
+		}
+
+		$prompt = sprintf( 'Compare the following business description: "%s" with the image titles listed in the JSON below: if the image is not aligned with what the business is selling, delete the description from the list. Do not include any explanations or introductions to the response. The JSON is: %s', $business_description, wp_json_encode( $image_titles ) );
+
+		$response = $ai_connection->fetch_ai_response( $token, $prompt, 30 );
+
+		if ( is_wp_error( $response ) || ! isset( $response['completion'] ) ) {
+			return $returned_images;
+		}
+
+		$filtered_image_titles = json_decode( $response['completion'] );
+
+		if ( ! is_array( $filtered_image_titles ) ) {
+			$response = $ai_connection->fetch_ai_response( $token, $prompt, 30 );
+
+			if ( is_wp_error( $response ) || ! isset( $response['completion'] ) ) {
+				return $returned_images;
+			}
+
+			$filtered_image_titles = json_decode( $response['completion'] );
+		}
+
+		if ( ! is_array( $filtered_image_titles ) ) {
+			return new \WP_Error( 'ai_service_unavailable', __( 'AI Service is unavailable, try again later.', 'woo-gutenberg-products-block' ) );
+		}
+
+		foreach ( $returned_images as $returned_image ) {
+			if ( isset( $returned_image['title'] ) && ! in_array( $returned_image['title'], $filtered_image_titles, true ) ) {
+				unset( $returned_image );
+			}
+		}
```

Comment:
I would add a comment to explain which kind of check we are doing.

Reply:
Done! https://github.com/woocommerce/woocommerce-blocks/pull/11759/commits/0ba01c8e7850d49364c3e84850319fbbd2305b84